Automattic/woocommerce-payments
# Add an endpoint to create a terminal intent

Fixes #3431

#### Changes proposed in this Pull Request

A new endpoint that leverages an existing API method to create a server-side intent for an in-person (or terminal) payment. It is specific to terminal payments and DOES NOT offer a way to create an open-ended payment intent.

#### Testing instructions

* Create a new order in WooCommerce with "Cash on delivery" as the payment method.
* Make a POST request to `/payments/orders/<order_id>/create_terminal_intent`. Refer the [docs](https://github.com/woocommerce/woocommerce/wiki/Getting-started-with-the-REST-API) to set up authorization correctly.
* You should get a successful (200) response with the id of the created intent.
* Try again with a non-existent order id. In this case, you should receive a 404 response with an appropriate error message.

-------------------

- [ ] Added changelog entries to both `readme.txt` and `changelog.txt`

**Post merge**

- [ ] Added testing instructions to the [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) (does not apply)
